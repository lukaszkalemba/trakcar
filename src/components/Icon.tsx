import React from 'react';
import SVG from 'react-inlinesvg';
import styles from './Icon.module.scss';

const Icon: React.FC<IconProps> = ({ src, ...props }) => {
  return (
    <div className={styles.icon} {...props}>
      <SVG src={src} />
    </div>
  );
};

interface IconProps {
  src: string;
}

export default Icon;
