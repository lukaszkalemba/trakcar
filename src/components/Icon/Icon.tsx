import React from 'react';
import SVG from 'react-inlinesvg';
import cx from 'classnames';
import styles from './Icon.module.scss';

const Icon: React.FC<IconProps> = ({ src, className, ...props }) => {
  return (
    <div className={cx(styles.icon, className)} {...props}>
      <SVG src={src} />
    </div>
  );
};

interface IconProps {
  src: string;
  className?: string;
}

export default Icon;
