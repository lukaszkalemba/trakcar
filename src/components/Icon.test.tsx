import React from 'react';
import { render } from '@testing-library/react';
import logo from 'assets/svgs/logo_trakcar.svg';
import Icon from './Icon';

describe('<Icon />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Icon src={logo} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
