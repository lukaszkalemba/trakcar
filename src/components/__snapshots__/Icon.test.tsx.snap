// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> matches snapshot 1`] = `
<div
  class="icon"
/>
`;
